### PR TITLE
ANTHA-2742 : Improve prompt aggregation

### PR DESCRIPTION
**Pull request by haydnKing**
Prompt aggregation was highly expensive for more than a handful of prompts, this algorithm change makes it much faster.

Should have no external effects. Am running antha-tester now to be on the safe side.

### CODE REVIEW

**Comment by haydnKing** [comment]
`antha-tester` passed happily as expected

File: microArch/scheduler/liquidhandling/executionhelpers.go [hunk 1]
```diff
@@ -97,33 +97,12 @@ func thisNode(ic *wtype.IChain, n graph.Node, tg graph.Graph, deps map[graph.Nod
...
-func related(i1, i2 *wtype.LHInstruction, topolGraph graph.Graph) bool {
-	if ancestor(graph.Node(i1), graph.Node(i2), topolGraph) || ancestor(graph.Node(i2), graph.Node(i1), topolGraph) {
-		return true
-	}
-
-	return false
-}
-
-func canAggHere(ar []*wtype.LHInstruction, ins *wtype.LHInstruction, topolGraph graph.Graph) bool {
+// unreachable return true ins cannot reach, or cannot be reached from, any of ar
```
Comment (msackman-synthace) [inline microArch/scheduler/liquidhandling/executionhelpers.go]: english has gone wrong here!

File: microArch/scheduler/liquidhandling/executionhelpers.go [hunk 2]
```diff
@@ -97,33 +97,12 @@ func thisNode(ic *wtype.IChain, n graph.Node, tg graph.Graph, deps map[graph.Nod
...
+func unreachable(ar []*wtype.LHInstruction, ins *wtype.LHInstruction, reachability graph.Reachability) bool {
```
Comment (msackman-synthace) [inline microArch/scheduler/liquidhandling/executionhelpers.go]: hmm, very minor, but just as I don't like negatives in flag names, here I think the `un` is better off lost, and then in the callsite, `!reachable(...)`.

Comment (haydnKing) [reply microArch/scheduler/liquidhandling/executionhelpers.go]: yeah fair enough, otherwise at some point someone's bound to write `!unreachable(...)`

**Pull request merged**